mkosi

```diff
@@ -268,59 +271,134 @@ def prepare_esp(args, loopdev):
 
     print_step("Formatting ESP partition completed.");
 
-def mkfs_ext4(label, mount, loopdev, partno):
-    subprocess.run(["mkfs.ext4", "-L", label, "-M", mount, partition(loopdev, partno)], check=True)
+def mkfs_ext4(label, mount, dev):
+    subprocess.run(["mkfs.ext4", "-L", label, "-M", mount, dev], check=True)
+
+def prepare_luks(args):
+
+    print_step("LUKS setup")
+    passphrase = luks_read_passphrase()
+
+    keys = {}
+    if args.luks_encryption == "all":
+        keys["root"] = passphrase
+        keys["home"] = passphrase
+        keys["srv"] = passphrase
+    elif args.luks_encryption == "data":
+        keys["home"] = passphrase
+        keys["srv"] = passphrase
+
+    return keys
+
+def luks_read_passphrase():
+    try:
+        passphrase_mode = os.stat('mkosi.passphrase').st_mode & (stat.S_IRWXU | stat.S_IRWXG | stat.S_IRWXO)
```

Comment:
not that it matters, but I never found the S_IRWXU constants readable... I'd always write this in octal...